pkg/apis/configs/broker/store.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+import (
+	"context"
+
+	"knative.dev/pkg/configmap"
+)
+
+type brDeliveryCfgKey struct{}
+
+// Config holds the collection of configurations that we attach to contexts.
+// +k8s:deepcopy-gen=false
+type Config struct {
+	BrokerDeliverySpecDefaults *Defaults
+}
+
+// FromContext extracts a Config from the provided context.
+func FromContext(ctx context.Context) *Config {
+	x, ok := ctx.Value(brDeliveryCfgKey{}).(*Config)
+	if ok {
+		return x
+	}
+	return nil
+}
+
+// FromContextOrDefaults is like FromContext, but when no Config is attached it
+// returns a Config populated with the defaults for each of the Config fields.
+func FromContextOrDefaults(ctx context.Context) *Config {
+	if cfg := FromContext(ctx); cfg != nil {
+		return cfg
+	}
+	defaults, _ := NewDefaultsConfigFromMap(map[string]string{})
+	return &Config{
+		BrokerDeliverySpecDefaults: defaults,
+	}
```

Comment:
It seems `defaults` will always be `nil`?

Reply:
You're right. Definitely seems strange to me, but this is how each FromContextsOrDefaults function is written in the rest of Knative, even when nil is guaranteed output. I just followed the standard without realizing, but I can change it to be more succinct.

Reply:
Done.